instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterProperties.java

```diff
@@ -20,6 +20,9 @@
  * <p>Get Exporter Endpoint
  *
  * <p>Get max wait time for Collector to process Span Batches
+ *
+ * <p>Note that this class mostly exists to support auto-completion in IDEs. The actual properties
```

Comment:
About the auto-completion of properties resolved from the `Environment` object, the Spring Boot code seems to add them to an `additional-spring-configuration-metadata.json` file. We could perhaps stay close to the way Spring Boot is implemented. I started an `additional-spring-configuration-metadata.json` file [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/4348a3e258751c69002c530b5495ebe76806b667/instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json).

Reply:
This file is recommended to be generated automatically from configuration properties files.

Reply:
> This file is recommended to be generated automatically from configuration properties files.

it looks like we are already doing this? (#8516)

Reply:
See https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor.adding-additional-metadata 

When a property is retrieved from a `Resource` object, Spring annotation processor can't automatically generate an autocompletion file. In this case, the practice in the Spring Boot project seems to add properties in the `additional-spring-configuration-metadata.json` file, example with `org.springframework.boot.context.logging.LoggingApplicationListener`: https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot/src/main/resources/META-INF/additional-spring-configuration-metadata.json

Reply:
> See https://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html#appendix.configuration-metadata.annotation-processor.adding-additional-metadata
> 
> When a property is retrieved from a `Resource` object, Spring annotation processor can't automatically generate an autocompletion file. In this case, the practice in the Spring Boot project seems to add properties in the `additional-spring-configuration-metadata.json` file, example with `org.springframework.boot.context.logging.LoggingApplicationListener`: https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot/src/main/resources/META-INF/additional-spring-configuration-metadata.json

A detailed example: 
 * https://github.com/spring-projects/spring-boot/blob/ff97fdc203290b55f010a2ee7263a0be34159b3f/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/logging/LoggingApplicationListener.java#L310
 * https://github.com/spring-projects/spring-boot/blob/ff97fdc203290b55f010a2ee7263a0be34159b3f/spring-boot-project/spring-boot/src/main/resources/META-INF/additional-spring-configuration-metadata.json#L10

Reply:
OK, then we should apply the best practice :smile: